manager/integration/tests/test_node.py

```diff
@@ -2239,4 +2256,282 @@ def test_node_eviction():
+            assert replica.mode == "RW"
+
+    client.update(node1, evictionRequested=True)
+
+    replica_node_soft_anti_affinity_setting = \
+        client.by_id_setting(SETTING_REPLICA_NODE_SOFT_ANTI_AFFINITY)
+    try:
+        client.update(replica_node_soft_anti_affinity_setting,
+                      value="true")
+    except Exception as e:
+        print("\nException when update "
+              "Replica Node Level Soft Anti-Affinity setting",
+              replica_node_soft_anti_affinity_setting)
+        print(e)
+
+    wait_for_volume_replica_count(client, volume1_name, 3)
+    wait_for_volume_replica_count(client, volume1_name, 2)
+
+    volume1 = client.by_id_volume(volume1_name)
+    for replica in volume1.replicas:
+        assert replica.hostId == node2.name
+        assert replica.running is True
+        assert replica.mode == "RW"
+
+    client.update(node3, allowScheduling=True, evictionRequested=False)
+    client.update(node1, allowScheduling=True)
+
+    replica_node_soft_anti_affinity_setting = \
+        client.by_id_setting(SETTING_REPLICA_NODE_SOFT_ANTI_AFFINITY)
+    try:
+        client.update(replica_node_soft_anti_affinity_setting,
+                      value="false")
+    except Exception as e:
+        print("\nException when update "
+              "Replica Node Level Soft Anti-Affinity setting",
+              replica_node_soft_anti_affinity_setting)
+        print(e)
+
+    client.update(node2, allowScheduling=False, evictionRequested=True)
+
+    volume1 = client.by_id_volume(volume1_name)
+    assert volume1.robustness == "healthy"
+
+    wait_for_volume_replica_count(client, volume1_name, 3)
+    wait_for_volume_replica_count(client, volume1_name, 2)
+
+    volume1 = client.by_id_volume(volume1_name)
+    assert volume1.robustness == "healthy"
+
+    wait_for_volume_replica_count(client, volume1_name, 3)
+    wait_for_volume_replica_count(client, volume1_name, 2)
+
+    volume1 = client.by_id_volume(volume1_name)
+    assert volume1.robustness == "healthy"
+
+    for replica in volume1.replicas:
+        assert replica.hostId != node2.name
+        assert replica.running is True
+        assert replica.mode == "RW"
+
+    v1md5sum = get_pod_data_md5sum(core_api,
+                                   pod1_name,
+                                   volume1_data_path)
+
+    assert v1md5sum == volume1_md5sum
+    client.update(node2, allowScheduling=True, evictionRequested=False)
+    volume1 = client.by_id_volume(volume1_name)
+
+    for replica in volume1.replicas:
+        if replica.hostId == node1.name:
+            break
+
+    volume1.replicaRemove(name=replica.name)
+
+    client.update(node3, allowScheduling=False, evictionRequested=True)
+    wait_for_volume_degraded(client, volume1_name)
+    wait_for_volume_healthy(client, volume1_name)
+
+    wait_for_volume_replica_count(client, volume1_name, 3)
+    wait_for_volume_replica_count(client, volume1_name, 2)
```

Comment:
Are we missing step 16? Check replicas on node-1 and node-2, and verify the data checksum? Thanks.

Reply:
Fixed, Thanks.